### ISSUE TITLE
Speed up WordNet tests and/or move to unit tests

### ISSUE THREAD

**Issue by kylepjohnson** [cltk/cltkv1#70]
I'm working on speeding up the testing process by breaking tests into two stages:
- doctests, which should build fast-ish
- unit tests, which contain slow-to-run tests

Since wordnet makes remote calls (and because it is so thoroughly covered -- good!), it takes ~5 mins to run.

I propose that you guys work on one or more of the following:
- refactoring the many little doctests ([example](https://github.com/cltk/cltkv1/blob/master/src/cltkv1/wordnet/wordnet.py#L98)) and moving them "up" into the class docstring. Would doing this reduce the number of remote calls you'd need to make?
- If the cannot be sped up with the above, them move most to a module `tests/test_wordnet.py`. These we'll as a more thorough "integration" test which we may or may not need during day-to-day development. In these, I really don't care about speed at all!

**Comment by todd-cook** [comment]
Elsewhere I have been able to keep doctests in place, but skip them when they take too long (e.g. see https://github.com/todd-cook/ML-You-Can-Use/blob/master/mlyoucanuse/bert_fun.py#L45 )
This would allow us to keep tests in place for documentation continuity, and also place some in the integration test (without the skip switch).
I'll profile the tests and see if there are any hotspots or places we can aggregate for a win.

**Comment by kylepjohnson** [comment]
`  # doctest: +SKIP` -- this would work too

**Comment by wmshort** [comment]
Is there a task I can take on without interfering with Todd's work on the tests? Todd you had mentioned some issues with the tests relating to the API -- shall I focus on these?

**Comment by todd-cook** [comment]
Yes, please look into those issues. If there are changes to be made, I'll
be able to move things around, and nothing will get lost.

**Comment by wmshort** [comment]
OK, perfect, I will get that done tonight or tomorrow morning! Thanks for all your help

William Michael Short
Lecturer in Classics & Ancient History
University of Exeter
Phone: 01392 724794
E-mail: w.short@exeter.ac.uk
Amory Building, Rennes Drive, Exeter, Devon, EX4 4RJ
Ancient Greek WordNet<https://greekwordnet.chs.harvard.edu/> | Latin WordNet<https://latinwordnet.exeter.ac.uk/> | Cylleneus: Next-gen corpus search for ancient languages<https://github.com/cylleneus>
[http://www.exeter.ac.uk/codebox/email-sig/staff-sig.gif]
This email and any attachment may contain information that is confidential, privileged, or subject to copyright, and which may be exempt from disclosure under applicable legislation. It is intended for the addressee only. If you received this message in error, please let me know and delete the email and any attachments immediately. The University will not accept responsibility for the accuracy/completeness of this email and its attachments.
